retry/retry_test.go

```diff
@@ -0,0 +1,147 @@
+package retry_test
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/go-quicktest/qt"
```

Comment:
why? experiment? I am a bit confused here, don't we change this test package two often now...

What about `gotest.tools/v3/assert`? If we introduce the new one, shouldn't we make it consistent everywhere....

Reply:
Good question. In my experiments I found "github.com/go-quicktest/qt" better than (the already good) "gotest.tools/v3/assert", for two reasons: it uses generics, so some misuses are caught at compile time, and it prints clearer diffs in certain cases.

You are right that we should be using only one (in the same spirit of log/slog). I will do the replacement, but as a subsequent PR.